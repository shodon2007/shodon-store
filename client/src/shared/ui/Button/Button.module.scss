.Button {
	padding: 10px 15px;
	border-radius: 5px;
	border: none;
	cursor: pointer;
	transition: all 0.3s ease;
	background-color: var(--primary-color);
	color: var(--bg-color);
}

.Button.smallest {
	padding: 5px 10px;
	font-size: var(--font-size-s);
	background-color: color-mix(in srgb, var(--primary-color) 50%, var(--bg-color));
}

.Button:hover {
	background-color: var(--primary-focus-color);
}
