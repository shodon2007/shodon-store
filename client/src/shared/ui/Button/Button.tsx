import { ButtonHTMLAttributes, FC } from "react";

import classNames from "src/shared/lib/classNames";

import cls from "./Button.module.scss";

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
    className?: string;
}

const Button: FC<ButtonProps> = ({ children, className, ...props }) => {
    return (
        <button {...props} className={classNames(cls.Button, {}, [className])}>
            {children}
        </button>
    );
};
export default Button;