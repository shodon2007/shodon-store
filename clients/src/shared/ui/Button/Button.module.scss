.Button {
	padding: 10px 15px;
	border-radius: 5px;
	border: none;
	cursor: pointer;
	transition: all 0.3s ease;
}

.Button:hover {
	background-color: var(--primary-focus-color);
}
