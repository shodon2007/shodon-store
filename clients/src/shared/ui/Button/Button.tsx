import { Link } from "react-router-dom";
import { ButtonHTMLAttributes, FC } from "react";

import classNames from "/shared/lib/classNames";

import cls from "./Button.module.scss";

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
    className?: string;
}

const Button: FC<ButtonProps> = ({ children, className, ...props }) => {
    return (
        <button {...props} className={classNames(cls.Button, {}, [className])}>
            {children}
        </button>
    );
};
export default Button;