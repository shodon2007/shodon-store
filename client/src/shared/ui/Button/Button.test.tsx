import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Button from './Button';

test('testing button tsx', () => {
    render(<Button>hello world</Button>);
    const buttonElem = screen.getByText('hello world');
    expect(buttonElem).toBeInTheDocument();
    expect(buttonElem).toMatchSnapshot();
});