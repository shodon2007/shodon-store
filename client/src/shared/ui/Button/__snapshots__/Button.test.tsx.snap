// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing button tsx 1`] = `
<button
  class="Button "
>
  hello world
</button>
`;
